### PR TITLE
RD-2427 related terms endpoint

### PR DESCRIPTION
**Pull request by iredpath**

### CODE REVIEW

File: rosette_apiUnitTests/rosette_apiUnitTests.cs [hunk 1]
```diff
@@ -1156,46 +1156,104 @@ public void TextEmbeddingTestFull()
...
+            Dictionary<string, string> responseHeaders = new JavaScriptSerializer().Deserialize<Dictionary<string, string>>(headersAsString);
+            Dictionary<string, object> content = new Dictionary<string, object>() {
+                {"similarTerms", terms}
+            };
+            SimilarTermsResponse expected = new SimilarTermsResponse(terms, responseHeaders, content, null);
+            String mockedContent = expected.ContentToString();
+            HttpResponseMessage mockedMessage = MakeMockedMessage(responseHeaders, HttpStatusCode.OK, mockedContent);
+            _mockHttp.When(_testUrl + "semantics/similar").Respond(req => mockedMessage);
+            SimilarTermsResponse response = _rosetteApi.SimilarTerms("spy");
+            Assert.AreEqual(expected, response);
```
Comment (guruxu) [inline rosette_apiUnitTests/rosette_apiUnitTests.cs:1222]: travis test is failing here

File: rosette_apiUnitTests/rosette_apiUnitTests.cs [hunk 2]
```diff
@@ -1156,46 +1156,104 @@ public void TextEmbeddingTestFull()
...
+            Init();
+            IDictionary<string, List<SimilarTerm>> terms = new Dictionary<string, List<SimilarTerm>>() {
+                {"eng", new List<SimilarTerm>() {new SimilarTerm("spy", 1.0m)}}
+            };
+            string headersAsString = " { \"Content-Type\": \"application/json\", \"date\": \"Thu, 11 Aug 2016 15:47:32 GMT\", \"server\": \"openresty\", \"strict-transport-security\": \"max-age=63072000; includeSubdomains; preload\", \"x-rosetteapi-app-id\": \"1409611723442\", \"x-rosetteapi-concurrency\": \"50\", \"x-rosetteapi-request-id\": \"d4176692-4f14-42d7-8c26-4b2d8f7ff049\", \"content-length\": \"72\", \"connection\": \"Close\" }";
+            Dictionary<string, string> responseHeaders = new JavaScriptSerializer().Deserialize<Dictionary<string, string>>(headersAsString);
+            Dictionary<string, object> content = new Dictionary<string, object>() {
+                {"similarTerms", terms}
+            };
+            SimilarTermsResponse expected = new SimilarTermsResponse(terms, responseHeaders, content, null);
```
Comment (guruxu) [inline rosette_apiUnitTests/rosette_apiUnitTests.cs:1217]: Ah, @iredpath I think `content` needs to be a `string` not `Dictionary` here.

Comment (iredpath) [reply rosette_apiUnitTests/rosette_apiUnitTests.cs:1217]: Assuming I'm not reading the assert error response backwards (which would make this an even more confusing error), I think the actual issue is with the mocked response.  Even though I'm fairly sure I checked that `mockedContent` looks correct, by the time it comes back in the mocked response we lose the inner `terms` map.  I don't know if the issue is the `mockedMessage` itself or the mocked response as a whole, but maybe a (de)serialization issue?

Comment (guruxu) [reply rosette_apiUnitTests/rosette_apiUnitTests.cs:1217]: Sorry my eyes were crossed. I'm more noob than you when comes to c#. There's on other suspect:

https://github.com/rosette-api/csharp/blob/12444107b7256a1d7ba4d10c619d1066fba84e31/rosette_api/SimilarTermsResponse.cs#L45-L46

would the default `content = null` wipe out something?

Comment (iredpath) [reply rosette_apiUnitTests/rosette_apiUnitTests.cs:1217]: Doesn't look like it, I just gave it a shot and no change in the job.

**Pull request merged**